docs/pbc/all/miscellaneous/202407.0/extend-the-audit-log-structure.md

```diff
@@ -0,0 +1,252 @@
+use Symfony\Component\HttpFoundation\Request;
+
+class CustomerBusinessUnitProcessorPlugin extends AbstractPlugin implements LogProcessorPluginInterface
+{
+    /**
+     * @param array<string, mixed> $data
+     *
+     * @return array<string, mixed>
+     */
+    public function __invoke(array $data): array
+    {
+        $customerTransfer = $this->findCurrentCustomer();
+
+        if (!$customerTransfer) {
+            return $data;
+        }
+
+        $currentRequestData = $this->getCurrentRequestData($customerTransfer);
+
+        if (isset($data['extra']['request'])) {
+            $data['extra']['request'] = array_merge(
+                $data['extra']['request'],
+                $currentRequestData,
+            );
+
+            return $data;
+        }
+
+        $data['extra']['request'] = $currentRequestData;
+
+        return $data;
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\CustomerTransfer|null
+     */
+    protected function findCurrentCustomer(): ?CustomerTransfer
+    {
+        $currentRequest = $this->getFactory()->getRequestStackService()->getCurrentRequest();
+
+        if (!$currentRequest || !$currentRequest->hasSession()) {
+            return null;
+        }
+
+        return $this->findCustomerInRequest($currentRequest);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CustomerTransfer $customerTransfer
+     *
+     * @return array<string, mixed>
+     */
+    protected function getCurrentRequestData(CustomerTransfer $customerTransfer): array
+    {
+        $currentRequestData = [];
+
+        $currentRequestData['customer_business_unit_name'] = $customerTransfer->getCompanyUserTransferOrFail()
+            ->getCompanyBusinessUnitOrFail()
+            ->getName();
+
+        return $currentRequestData;
+    }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     *
+     * @return \Generated\Shared\Transfer\CustomerTransfer|null
+     */
+    protected function findCustomerInRequest(Request $request): ?CustomerTransfer
+    {
+        return $request->getSession()->get('customer data');
+    }
+}
+```
+
+</details>
+
+4. Register the newly introduced plugin for the `security` log type for the `Yves` application:
+
+**src/Pyz/Yves/Log/LogDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Yves\Log;
+
+use Spryker\Yves\Kernel\Container;
+use Spryker\Yves\Log\LogDependencyProvider as SprykerLogDependencyProvider;
+
+class LogDependencyProvider extends SprykerLogDependencyProvider
+{
+    /**
+     * @return list<\Spryker\Shared\Log\Dependency\Plugin\LogProcessorPluginInterface>
+     */
+    protected function getYvesSecurityAuditLogProcessorPlugins(): array
+    {
+        return [
+            // other plugins
+            new CustomerBusinessUnitProcessorPlugin(),
+        ];
+    }
+}
+```
+Now when you log in as a customer with a business unit, the corresponding business unit name will be added to the log data.
```

Comment:
I'd prefer to have a link to the customer, who did the action, so I can check there any details.
Did you think about this?

Reply:
The core functionality extends data with customer email/reference OOTB